src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -135,6 +139,200 @@ void RaiseSecond()
             VisualStudio.Editor.Verify.TextContains("Second?.");
         }
 
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.EditorConfig)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        [WorkItem(15003, "https://github.com/dotnet/roslyn/issues/15003")]
+        [WorkItem(19089, "https://github.com/dotnet/roslyn/issues/19089")]
+        public void ApplyEditorConfigAndFixAllOccurrences()
+        {
+            var markup = @"
+class C
+{
+    public int X1
+    {
+        get
+        {
+            $$return 3;
+        }
+    }
+
+    public int Y1 => 5;
+
+    public int X2
+    {
+        get
+        {
+            return 3;
+        }
+    }
+
+    public int Y2 => 5;
+}";
+            var expectedText = @"
+class C
+{
+    public int X1 => 3;
+
+    public int Y1 => 5;
+
+    public int X2 => 3;
+
+    public int Y2 => 5;
+}";
+
+            // CodingConventions only sends notifications if a file is open for all directories in the project
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), @"Properties\AssemblyInfo.cs");
+
+            // Switch back to the main document we'll be editing
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "Class1.cs");
+
+            /*
+             * The first portion of this test adds a .editorconfig file to configure the analyzer behavior, and verifies
+             * that diagnostics appear automatically in response to the newly-created file. A fix all operation is
+             * applied, and the result is verified against the expected outcome for the .editorconfig style.
+             */
+
+            MarkupTestFile.GetSpans(markup, out var text, out ImmutableArray<TextSpan> spans);
+            SetUpEditor(markup);
+            VisualStudio.WaitForApplicationIdle(CancellationToken.None);
+            VisualStudio.Editor.Verify.CodeActionsNotShowing();
+
+            var editorConfig = @"root = true
+
+[*.cs]
+csharp_style_expression_bodied_properties = true:warning
+";
+
+            VisualStudio.SolutionExplorer.AddFile(new ProjectUtils.Project(ProjectName), ".editorconfig", editorConfig, open: false);
+
+            // We have no way of waiting on the CodingConventions library to send its file system changed notifications
+            Thread.Sleep(TimeSpan.FromSeconds(1));
+
+            VisualStudio.WaitForApplicationIdle(CancellationToken.None);
+            VisualStudio.Workspace.WaitForAllAsyncOperations(
+                FeatureAttribute.Workspace,
+                FeatureAttribute.SolutionCrawler,
+                FeatureAttribute.DiagnosticService);
+            VisualStudio.Editor.InvokeCodeActionList();
+            VisualStudio.Editor.Verify.CodeActions(
```

Comment:
shouldn't this only check what you are looking for exist rather whole list? what happen if we add new code action?

Reply:
Originally it was written in this form so I could include assertions about the _absence_ of certain items. I'll see if the test can be reduced without leaving it open for false positives.